Fixing issues with column sizes

OlivierBBB opened the pull request.

OlivierBBB committed:
fix: blockdata

OlivierBBB committed:
ras: static imports

OlivierBBB committed:
fix: stp column size related changes

OlivierBBB committed:
ras: isCreate and isCall methods for OpCodeData

OlivierBBB committed:
ras: gasUsed is now long

OlivierBBB committed:
fix: accumulated gas now long

This fixes an issue where casting it to (int) left us with a remainder
mod some power of 2 instead of the actual accumulatedGasUsedInBlock

OlivierBBB committed:
feat: executionPaused and lastValidGasNext in CallFrame to allow us to fill GasExpected correctly after resuming execution in a frame

OlivierBBB committed:
fix: more fixes related to gas now being a long

OlivierBBB committed:
fix: expectedGas now gets the lastValidGasNext() when execution resumes

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/opcode/OpCodeData.java:
```diff
@@ -77,6 +77,14 @@ public boolean isHalt() {
     return this.instructionFamily == InstructionFamily.HALT;
   }
 
+  public boolean isCall() {
```
We already have those method somewhere else (based on the opcode, not the opcode family)

OlivierBBB replied:
Yes. But it makes more sense to have them defined on the instruction family level.

OlivierBBB replied:
I moved these methods to the `OpCodeData` level.

OlivierBBB committed:
Merge branch 'arith-dev' into 1392-fixing-issues-with-column-sizes

OlivierBBB committed:
ras: int -> long in RamToExoWithPadding.java

OlivierBBB committed:
fix: OpCode related fixes

- removed pushFlag from opcodes.yml
- removed jumpFlag from opcodes.yml (which BTW was a misnomer)
- make the isXxx() methods of OpCode.java return the same method from
OpCodeData.java

OlivierBBB committed:
spotless

OlivierBBB committed:
Merge branch 'arith-dev' into 1392-fixing-issues-with-column-sizes
